src/hooks/useSignupStatus.ts

```diff
@@ -0,0 +1,48 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import { commonFetch } from '@openshift/dynamic-plugin-sdk-utils';
+
+export enum UserSignupStatus {
+  NOT_SIGNEDUP = 'NotSignedup',
+  PENDING_APPROVAL = 'PendingApproval',
+  PROVISIONED = 'Provisioned',
+}
+
+export const useSignupStatus = (): [string, Dispatch<SetStateAction<string>>, boolean] => {
```

Comment:
Tiny nit: just for stronger typing
```suggestion
export const useSignupStatus = (): [UserSignupStatus, Dispatch<SetStateAction<string>>, boolean] => {
```